cmd/container-manager/manage_ctr_mgr.sh

```diff
@@ -0,0 +1,323 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+DOCKER_BIN=docker
+SCRIPT_PATH=$(dirname "$(readlink -f "$0")")
+SCRIPT_NAME=${0##*/}
+source "${SCRIPT_PATH}/../../.ci/lib.sh"
+source /etc/os-release
+
+force=false
+ctr_manager=""
+subcommand=""
+runtime=""
+tag=""
+
+get_cc_versions > /dev/null 2>&1
+
+usage(){
+	cat << EOF
+This script helps you install the correct version of docker
+to use with Clear Containers.
+WARNING: This tool will overwrite any docker configuration that you may have
+modified.
+Usage: $SCRIPT_NAME [docker] [install|info|configure|remove] <options>
+Options:
+	-f                         : Force install action (even if another version is installed)
+	-h                         : Help, show this information.
+	-r <runtime>               : Supported runtimes: runc and cc-runtime
+	-t <tag>                   : Tags supported: swarm, latest. If you do not specify
+                                     a tag, the script will use latest as default.
+                                     With this tag you can install the correct version
+                                     of docker that has CC compatibility with swarm.
+Example:
+	./$SCRIPT_NAME docker install -t swarm -f
+EOF
+}
+
+die(){
+	msg="$*"
+	echo "$SCRIPT_NAME - ERROR: $msg" >&2
+	return 1
+}
+
+warning(){
+	msg="$*"
+	echo "$SCRIPT_NAME - WARNING: $msg" >&2
+}
+
+message(){
+	msg="$*"
+	echo "$SCRIPT_NAME - INFO: $msg" >&2
+}
+
+log_message(){
+	message="$1"
+	logger -s -t "$(basename $0)" "$message"
+}
+
+parse_subcommand_options(){
+	while getopts ":fr:t:" opt; do
+		case $opt in
+			f)
+				force=true
+				;;
+			r)
+				runtime="${OPTARG}"
+				;;
+			t)
+				tag="${OPTARG}"
+				;;
+			\?)
+				echo "Invalid option: -${OPTARG}" >&2
+				usage
+				exit 1
+		esac
+	done
+}
+
+# This function handles the installation of the required docker version.
+install_docker(){
+	# Check if docker is present in the system
+	if [ "$(info_docker)" ] && [ ${force} == false ]; then
+		die "Docker is already installed. Please use -f flag to force new installation"
+	elif [ "$(info_docker)" ] && [ ${force} == true ]; then
+		remove_docker
+	fi
+
+	if [ -z "$tag" ] || [ "$tag" == "latest" ] ; then
+		# If no tag is recevied, install latest compatible version
+		log_message "Installing docker $docker_version"
+		docker_version=${docker_version/v}
+		docker_version=${docker_version/-*}
+		if [ "$ID" == "ubuntu" ]; then
+			pkg_name="docker-ce=${docker_version}.0~ce-0~ubuntu"
+			repo_url="https://download.docker.com/linux/ubuntu"
+			sudo -E apt-get -y install apt-transport-https ca-certificates software-properties-common
+			curl -fsSL "${repo_url}/gpg" | sudo apt-key add -
+			sudo -E add-apt-repository "deb [arch=amd64] ${repo_url} $(lsb_release -cs) stable"
+			sudo -E apt-get update
+			sudo -E apt-get -y install docker-ce
```

Comment:
This will install the *newest* version of docker as `pkg_name` isn't being used here.

Also, I wonder how reliable those `pkg_name` variables will be. It might be safer to add the repo URL, do the `apt-get update`, then do something like:

```
docker_version_full=$(apt-cache show docker-ce|grep ^Version:|awk '{print $2}')
sudo -E apt-get -y install "$docker_version_full"
```